includes/fraud-prevention/class-buyer-fingerprinting-service.php

```diff
@@ -66,22 +51,18 @@ public static function set_instance( self $instance = null ) {
 	 * @return string Hashed data.
 	 */
 	public function hash_data_for_fraud_prevention( string $data ): string {
-		return hash( 'sha512', $data );
+		return hash( 'sha512', $data, false );
 	}
 
 	/**
 	 * Returns fraud prevention data for an order.
 	 *
-	 * @param int $order_id The WC order id.
-	 *
 	 * @return string[] An array of hashed data for an order.
 	 */
-	public function get_hashed_data_for_order( int $order_id ): array {
-		$order = wc_get_order( $order_id );
-
+	public function get_hashed_data_for_customer(): array {
 		return [
-			'shopper_ip_hash'        => self::hash_data_for_fraud_prevention( $order->get_customer_ip_address() ),
-			'shopper_useragent_hash' => self::hash_data_for_fraud_prevention( strtolower( $order->get_customer_user_agent() ) ),
+			'fraud_prevention_data_shopper_ip_hash' => $this->hash_data_for_fraud_prevention( WC_Geolocation::get_ip_address() ),
```

Comment:
In the future it'd be great for IP and User-Agent to be collected with JS from the browser client with the current implementation as a fallback.

Reply:
With this change singleton becomes useful for testing. E.g. you add two one-line methods to  `Buyer_Fingerprinting_Service`, i.e. `get_customer_ip` and `get_customer_useragent`. In the tests you create a mock-class and override these two methods with whatever values you need.